ideascube/mediacenter/management/commands/remove_medias.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+import os
+import sys
+import argparse
+from os.path import join as pj
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+
+from ideascube.mediacenter.models import Document
+from ideascube.utils import printerr
+
+
+class Command(BaseCommand):
+    help = ('Remove files from the mediacenter.')
+
+    def add_arguments(self, parser):
+        subs = parser.add_subparsers(
+            title='Commands', dest='cmd', metavar='',
+            parser_class=argparse.ArgumentParser)
+        remove = subs.add_parser(
+            'clean_orphans',
+            help='Clean mediacenter files not associated with a document.')
+        remove.set_defaults(func=self.clean_orphans)
+        parser.add_argument('--dry-run', action='store_true',
+                            help='Print the list of medias that would be '
+                                 'removed. Do not actually remove them')
+
+    def abort(self, msg):
+        self.stderr.write(msg)
+        sys.exit(1)
+
+    def handle(self, *args, **options):
+        if 'func' not in options:
+            self.parser.print_help()
+            self.parser.exit(1)
+
+        options['func'](options)
+
+    def clean_orphans(self, options):
+        files_to_remove = self._get_orphan_files()
+        if options['dry_run']:
+            print("Files to remove are :")
+            for _file in files_to_remove:
+                print(" - '{}'".format(_file))
+        else:
+            for f in files_to_remove:
+                try:
+                    os.unlink(f)
+                except Exception as e:
+                    printerr("ERROR while deleting {}".format(f))
+                    printerr("Exception is {}".format(e))
+
+    def _get_orphan_files(self):
+        # List all (original and preview) files in the fs
+        original_files_root_dir = pj(settings.MEDIA_ROOT,
+                                     'mediacenter/document')
+        try:
+            _, _, original_files_in_fs = next(os.walk(original_files_root_dir))
+        except StopIteration:
+            original_files_in_fs = []
+        else:
+            original_files_in_fs = (pj(original_files_root_dir, f)
+                                    for f in original_files_in_fs)
+        files_in_fs = set(original_files_in_fs)
+
+        preview_files_root_dir = pj(settings.MEDIA_ROOT,
+                                    'mediacenter/preview')
+        try:
+            _, _, preview_files_in_fs = next(os.walk(preview_files_root_dir))
+        except StopIteration:
+            preview_files_in_fs = []
+        else:
+            preview_files_in_fs = (pj(preview_files_root_dir, f)
+                                   for f in preview_files_in_fs)
+        files_in_fs = files_in_fs.union(preview_files_in_fs)
```

Comment:
In case you didn't know, `set1.union(set2)` is strictly equivalent to `set1 | set2`, which then allows you to use `set1 |= set2`.

I personally like it, as it reminds me of the logical OR operator: the union of two sets is the result of OR-ing each of their elements.

(same with `set1 = set1.difference(set2)` and `set1 -= set2`)

But that's a personal preference. However if you prefer the longer form, can I suggest `set1.update(set2)` and `set1.difference_update(set2)`?